### PR TITLE
#620 | Sai/Meghna | Defines contract for POST API internal/cm

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

File: src/main/resources/static/api.yaml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          type: string
+        url:
+          type: string
+        cm_id:
+          type: string
+        suffix:
+          type: string
+        active:
+          type: boolean
+        blocklisted:
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml:91]: anyone ever register a CM with blacklisted and inactive?

File: src/main/resources/static/api.yaml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          type: integer
+          enum: [1000, 10001]
+        message:
+          type: string
+      xml:
+        name: Error
+
+
+
+
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml]: remove this.

File: src/main/resources/static/api.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+          type: string
+        active:
+          type: boolean
+        blocklisted:
+          type: boolean
+        license:
+          type: string
+        licensing_authority:
+          type: string
+      xml:
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml]: required block is missing where you are supposed to have suffix, id, url

File: src/main/resources/static/api.yaml (hunk 1 above)
Comment (MeghnaThoughtworks) [reply src/main/resources/static/api.yaml:91]: These two are in the consent-manager table so it should be true by default?

File: src/main/resources/static/api.yaml (hunk 3 above)
Comment (MeghnaThoughtworks) [reply src/main/resources/static/api.yaml]: done

File: src/main/resources/static/api.yaml (hunk 2 above)
Comment (MeghnaThoughtworks) [reply src/main/resources/static/api.yaml]: done

**Review by ganesanarun — APPROVED** [review]

**Pull request merged**